migen/build/lattice/platform.py

```diff
@@ -18,7 +18,13 @@ def __init__(self, *args, toolchain="diamond", **kwargs):
     def get_verilog(self, *args, special_overrides=dict(), **kwargs):
         so = dict(common.lattice_special_overrides)
         so.update(special_overrides)
-        return GenericPlatform.get_verilog(self, *args, special_overrides=so, **kwargs)
+        if isinstance(self.toolchain, diamond.LatticeDiamondToolchain):
+            return GenericPlatform.get_verilog(self, *args, special_overrides=so, **kwargs)
+        elif isinstance(self.toolchain, icestorm.LatticeIceStormToolchain):
+            return GenericPlatform.get_verilog(self, *args, special_overrides=so,
+                attr_translate=self.toolchain.attr_translate, **kwargs)
```

Comment:
So much boilerplate. Can't they just both support `attr_translate`?

Reply:
Refactored. Diamond now explicitly supports `DummyAttrTranslate` due to refactoring, which was the default anyway before this commit.